lumicks/pylake/detail/imaging_mixins.py

```diff
@@ -68,23 +80,34 @@ def extratags(timestamp_range):
             return (datetime,)
 
         # Save the tiff file page by page
-        with tifffile.TiffWriter(filename) as tif:
-            metadata = self._tiff_image_metadata()
-            exposure_times = (
-                np.atleast_1d(np.diff(np.vstack(frame_exposure_ranges), axis=1).squeeze()) * 1e-6
-            )
-            for frame, timestamp_range, exposure_time in zip(
-                frames, frame_timestamp_ranges, exposure_times
-            ):
-                metadata["Exposure time (ms)"] = exposure_time
-                tif.write(
-                    frame,
-                    contiguous=False,  # write tags on each page
-                    extratags=extratags(timestamp_range),
-                    metadata=None,  # suppress tifffile default ImageDescription tag
-                    description=json.dumps(metadata, indent=4),
-                    **self._tiff_writer_kwargs(),
+        try:
+            with tifffile.TiffWriter(filename, bigtiff=bigtiff) as tif:
+                metadata = self._tiff_image_metadata()
+                exposure_times = (
+                    np.atleast_1d(np.diff(np.vstack(frame_exposure_ranges), axis=1).squeeze())
+                    * 1e-6
                 )
+                for frame, timestamp_range, exposure_time in zip(
+                    frames, frame_timestamp_ranges, exposure_times
+                ):
+                    metadata["Exposure time (ms)"] = exposure_time
+                    tif.write(
+                        frame,
+                        contiguous=False,  # write tags on each page
+                        extratags=extratags(timestamp_range),
+                        metadata=None,  # suppress tifffile default ImageDescription tag
+                        description=json.dumps(metadata, indent=4),
+                        **self._tiff_writer_kwargs(),
+                    )
+        except Exception as e:
+            if "format requires 0 <= number <= 4294967295" in str(e):
+                raise ValueError(
```

Comment:
I'm wondering if it's a `ValueError` or if a `RuntimError` is better suited as the image to be exported is not passed as an argument the the `export_tiff` method.

Reply:
Yeah, I was doubting the exact same thing. I think it depends on whether you consider the error to be "oh no, the tiff turns out to big", or whether you consider the error to be the fact that `bigtiff` is erroneously set to `False`.

Reply:
I like the explanation that the user did not explicitly ask for a `bigtiff`. So no, in this sense `RuntimeError` is not more appropriate, and I would leave `ValueError` as it is, as it also implicitly conveys the message that the user can change the behavior by specifying a different value for `bigtiff`.

Reply:
Alright! Added a test, so we'd spot it if this upstream error ever changes! :)